### ISSUE TITLE
Chunk post-processing

### ISSUE THREAD

**Issue by lemoinem** [flowjs/flow.js#266]
**Context:**
In our web-application, after a file has been uploaded, we need to do some post-processing. It includes several additional calls to the server. We use the `fileSuccess` event to trigger this process.

**Problem:**
On slow/bad connections, some of our calls get queued and, eventually, timeout. This is because although flow.js queues the chunk uploads, there is no mechanism to keep an upload slot occupied while we do the post-processing. 

In the end, the post-processing requests compete with chunk upload. It fills up the browser queue and calls timeout.

This happens with user uploading 50+ files at once.

**Idea:**
One solution we were thinking about would be to add a post-processing feature to flow.js, similar to how chunk pre-processing is implemented.

Basically, when a chunk is uploaded, flowjs would check whether chunk post-processing is enabled. If yes, it will switch the chunk to postprocessing and wait until the chunk is marked as done. Chunk marked as postprocessing will count as still uploading in `_shouldUploadNext`.

We are ready to try and develop a PR for it if you are interested in merging it. What do you think?

Note: we are still looking into other avenues, but we'd like to know if adding this feature to flowjs via a PR is a possibility.

**Comment by AidasK** [comment]
If pull request is small and follows same code principals without any BREAKING CHANGES, then it's ok. Bring it on ;)
